WarehouseManagement.Core/Services/DeliveryService.cs

```diff
@@ -296,4 +297,139 @@ public async Task<ICollection<DeliveryDto>> GetAllDeletedAsync()
 
         return deliveries;
     }
+
+    public async Task<DeliveryHistoryDto> GetHistoryAsync(int id)
+    {
+        var delivery = await repository.GetByIdAsync<Delivery>(id);
+
+        if (delivery == null)
+        {
+            throw new KeyNotFoundException($"{DeliveryWithIdNotFound} {id}");
+        }
+
+        var relatedEntriesIds = repository
+            .AllReadOnly<Entry>()
+            .Where(e => e.DeliveryId == delivery.Id)
+            .Select(e => e.Id);
+
+        var changes = repository
+            .AllReadOnly<EntityChange>()
+            .Where(change => int.Parse(change.EntityId) == delivery.Id || relatedEntriesIds.Any(id => id == int.Parse(change.EntityId)));
+
+        var deliveryHistory = new DeliveryHistoryDto
+        {
+            Id = delivery.Id,
+            Changes = await changes.Select(change => new DeliveryHistoryDto.Change
+            {
+                EntityId = int.Parse(change.EntityId),
+                PropertyName = change.PropertyName,
+                From = change.OldValue!,
+                To = change.NewValue!,
+                Type = change.EntityName == "Delivery" ? DeliveryHistoryChangeType.Delivery : DeliveryHistoryChangeType.Entry,
+                LogType = change.PropertyName == "Status" && change.EntityName == "Entry" ? LogType.EntryStatusChange :
+                    change.PropertyName == "Status" && change.EntityName == "Delivery" ? LogType.DeliveryStatusChange :
+                    change.PropertyName == "ZoneId" ? LogType.ZoneChange : LogType.Split // May be refactored based on the Move functionality
+            }).ToListAsync()
+        };
+
+        return deliveryHistory;
+    }
+
+    public async Task ChangeDeliveryStatusIfNeeded(int id)
```

Comment:
where are we going to be calling this btw? I currently dont see it used in this PR